include/exadg/incompressible_navier_stokes/spatial_discretization/operator_dual_splitting.cpp

```diff
@@ -641,10 +630,13 @@ OperatorDualSplitting<dim, Number>::local_interpolate_velocity_dirichlet_bc_boun
         {
           AssertThrow(false, dealii::ExcMessage("Not implemented."));
         }
-
-        integrator.submit_dof_value(g, index);
+        // we do not need JxW so just write directly into values
+        auto * values = integrator.begin_values() + q;
+        for(unsigned int comp = 0; comp < integrator.n_components; ++comp)
+          values[comp * integrator.n_q_points] = g[comp];
       }
-
+      // integration reorders the integration points to the corresponding DoFs
+      integrator.integrate(dealii::EvaluationFlags::values);
```

Comment:
I think to a reader of the code it might not be entirely clear what kind of reordering is needed here. The important point is that, in a slight misuse of the computational interfaces, the "quadrature points" on the faces get used to embed the information into the cell DoFs by this call, to later be able to set the respective values on the cell. However, after having thought about the used concept I believe that this concept is extremely brittle: Think about the case of boundary conditions for two faces of an element. When we work on the second face, only the values along that face will be non-zero after `integrate`, the other values are zero, include the ones we tried to set on the other face. But we then call `set_dof_values`, thus overwriting the other values. For Navier-Stokes, this only affects Raviart-Thomas elements where only one component is zero (but I wonder if we take the Piola transformation properly into account), but for the structure and Poisson cases I definitely wonder if this is the right approach. I might misunderstand something here, but it definitely looks like a very dangerous approach to me.